### DISCUSSION TITLE
enrich ja3 query

### DISCUSSION THREAD

**Discussion by myViper** [Security-Onion-Solutions/securityonion#3566] [Unsupported Versions]
I recently wrote a query to pull ja3 values from bro-ssl, I'd like to display that with the ja3 user-agent pulled from https://ja3er.com/search/<ja3-md5>.  Problem is, I'm not sure how to do that.  Any pointers?  ja3er.com provides ajax to do this, but how would I use this within elastic?
The code:
$.getJSON( "https://ja3er.com/json", function( json ) {
  console.log( "JSON Data: " + json.ja3 );
 });

Any help is appreciated.

**Comment by weslambert** [comment]
I'll look into this and see if I can provide an example soon.  You would just want to be careful, as doing this directly in the pipeline has the potential for added latency.  Ideally, you would want to do this in post-processing or in a way that wouldn't block the main pipeline.

**Reply by myViper** [reply]
Thank you.  I did subsequently discover that Suricata will compute JA3 hashes and alert based on the result. Need to enable in suricata.yaml.  I'd still like to be able to do this in Elastic, if nothing else for the learning opportunity.

**Reply by presianbg** [reply]
This is awesome idea actually! Combine this with domainstats and we have some really nice hunting capabilities regardless of the TLS + SNI encryption.

Another possibility is to map the ja3 hash with the UA via the zeek intel plugin, but I'm not sure how to implement this in the best way possible.
https://ja3er.com/downloads.html

EDIT:
Looks like its already there:
https://github.com/salesforce/ja3/blob/master/zeek/intel_ja3.zeek

If we convert the json from ja3er into the proper intel format and add a plugin, which read from this ja3-UA-mapping.dat file, we can have it!

**Reply by myViper** [reply]
Yes, zeek computes ja3 already, easy enough to extract that.  
Suricata compares with this fingerprint database: https://github.com/trisulnsm/trisul-scripts/blob/master/lua/frontend_scripts/reassembly/ja3/prints/ja3fingerprint.json

**Reply by presianbg** [reply]
It computes ja3 and ja3s to enrich the ssl.log, but I'm wondering if the INTEL engine looks at those values.
I will try to create something similar to what suricata do and get back with the results ;)

**Reply by weslambert** [reply]
I know @bryant-treacle did some work with this to some degree a while back, and presented it at SOC '19:
https://youtu.be/-WD9BWlENwc?list=PLljFlTO9rB16UHGHbo4dTR5qzvJObEBoa&t=822

**Reply by presianbg** [reply]
Yes, thanks for the video, Its awesome!
Here are the scripts:
https://github.com/bryant-treacle/SSL_Traffic_Hunting

Can we have this build-in in near future releases?

**Reply by SanibelJack** [reply]
Stupid question. Would you enable the JA3 hashes in the global.sls or the yaml file?
